linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -631,8 +633,23 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
                   transformExprNoChar(rhs))
           }
 
-        case Assign(lhs @ (_:VarRef | Transient(JSVarRef(_, _)) |
-            _:SelectStatic | _:JSGlobalRef), rhs) =>
+        case Assign(select: SelectStatic, rhs) =>
+          // Destructure separately, otherwise 2.11 crashes.
+          val SelectStatic(className, item) = select
+
+          val scope = (className, item.name)
+
+          if (needToUseGloballyMutableVarSetter(scope)) {
+            unnest(rhs) { (rhs, env0) =>
+              implicit val env = env0
+              js.Apply(globalVar("u", scope), transformExprNoChar(rhs) :: Nil)
```

Comment:
Couldn't the static field have a `char` type? I think you need to use `transformExpr` with an expected type:
```suggestion
              js.Apply(globalVar("u", scope), transformExpr(rhs, select.tpe) :: Nil)
```